components/google_sheets/sources/new-spreadsheet/new-spreadsheet.js

```diff
@@ -0,0 +1,95 @@
+const common = require("../common");
+
+module.exports = {
+  ...common,
+  key: "google_sheets-new-spreadsheet",
+  name: "New Spreadsheet (Instant)",
+  description:
+    "Emits an event each time a new spreadsheet is created in a drive.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    folders: {
+      propDefinition: [
+        common.props.google_sheets,
+        "folders",
+        (c) => ({ driveId: c.watchedDrive }),
+      ],
+    },
+  },
+  hooks: {
+    ...common.hooks,
+    async deploy() {
+      this._setLastFileCreatedTime(Date.now());
+    },
+  },
+  methods: {
+    ...common.methods,
+    shouldProcess(file) {
+      const watchedFolders = new Set(this.folders);
+      return (
+        watchedFolders.size == 0 ||
+        file.parents.some((p) => watchedFolders.has(p))
+      );
+    },
+    async getSpreadsheetToProcess(event) {
+      const pageToken = this._getPageToken();
+      const {
+        changedFiles,
+        newStartPageToken,
+      } = await this.google_sheets.getChanges(
+        pageToken,
+        this.watchedDrive === "myDrive" ? null : this.watchedDrive
+      );
+      this._setPageToken(newStartPageToken);
+
+      const file = changedFiles
+        .filter((file) => file.mimeType.includes("spreadsheet"))
+        .shift();
+      if (file && this.shouldProcess(file)) return file;
```

Comment:
Shouldn't we return a collection of files? What if multiple spreadsheets were created since the last sync?
```suggestion
      return changedFiles
        .filter((file) => file.mimeType.includes("spreadsheet"))
        .filter(this.shouldProcess, this);
```